internal/handlers/llm_mcp_bridge.go

```diff
@@ -440,7 +440,17 @@ func (b *LLMMCPBridge) getClientForTool(toolName string) mcp.MCPClientInterface
 }
 
 // executeToolCall executes a detected tool call (using the new ToolCall struct)
-func (b *LLMMCPBridge) executeToolCall(ctx context.Context, toolCall *ToolCall) (string, error) {
+func (b *LLMMCPBridge) executeToolCall(ctx context.Context, toolCall *ToolCall, extraArgs map[string]interface{}) (string, error) {
+	for k, v := range extraArgs {
```

Comment:
This overwrites tool arguments. If a tool already has channel_id parameter, this will overwrite it. Should check for conflicts:
  ```
  if _, exists := toolCall.Args[k]; !exists {
      toolCall.Args[k] = v
  }
  ```